src/World.java

```diff
@@ -5,26 +5,46 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Random;
 import java.util.Scanner;
 
 public class World {
     // Map size in amount of hexagonal tiles in each dimension.
     public final int mapWidth = 31;
     public final int mapHeight = 34;
 
+    private Random random = new Random();
     private final City[] cities;
 
     //Retrieve file that stores world related info (Cities, Products, etc.)
     private Terrain[][] map;
 
     public World(List<Company> companies) {
         Log.debug("Generating world with " + companies.size() + " companies.");
+
+        cities = readCitiesFromFile("/assets/data/map-EasternUS.txt").toArray(new City[0]);
+        ArrayList<City> candidateStartCities = new ArrayList<>(Arrays.asList(cities));
+
+        Log.debug("\nStaring cities:");
+        // Handle this error condition better if possible.
```

Comment:
I think a better way would be to add a public static final field MAX_PLAYERS somewhere and throw a RuntimeException if there are fewer starting cities than that.
This situation can pretty much only be caused by a developer or modder setting things up wrong, so we want to fail fast and get it fixed.

Reply:
Done.